Assets/_BForBoss/_Weapons/Scripts/Weapons/CrossHairBehaviour.cs

```diff
@@ -6,13 +6,19 @@
 
 namespace Perigon.Weapons
 {
-    public class CrosshairBehaviour : MonoBehaviour
+    public interface ICrossHairProvider
```

Comment:
This is setup so that it's more of a dependency inversion. 
You pass in this component via inspector into the main WeaponManager and inject this into each weapon. 

As opposed to what it was before - drag and drop into each weapon in inspector, which seems not as sustainable.